components/common/CustomButton.vue

```diff
@@ -0,0 +1,137 @@
+<template>
+  <div class="button-wrapper">
+    <Component
+      :is="type"
+      class="button"
+      :class="{
+        'button--reverse': iconSuffix,
+        'button--animate-heart': icon === 'heart',
+      }"
+      :href="href"
+      :target="href && '__blank'"
+      :to="to"
+      role="link"
+      tabindex="0"
+    >
+      <span class="button__icon" role="presentation" aria-hidden="true">
+        <ArrowRight
+          v-if="icon === 'arrow-right'"
+          class="button__icon--arrow-right"
+        />
+        <Heart v-else-if="icon === 'heart'" class="button__icon--heart" />
+      </span>
+      <span class="button__text">
+        <slot />
+      </span>
+    </Component>
+  </div>
+</template>
+
+<script>
+import ArrowRight from '~/assets/svg/icons/arrow_right.svg?inline'
+import Heart from '~/assets/svg/icons/heart.svg?inline'
+
+export default {
+  components: {
+    ArrowRight,
+    Heart,
+  },
+  props: {
+    href: {
+      type: String,
+      default: null,
+    },
+    to: {
+      type: String,
+      default: null,
+    },
+    icon: {
+      type: String,
+      default: null,
+      validator: (value) => {
+        return ['arrow-right', 'heart'].includes(value)
```

Comment:
💬 validations 🤩! Do we need to accept the default value as well?

Reply:
Do you mean the `null`? That's the way to tell Vue is not a required property, not sure if you meant that. 😅